Sequence runner

ysoo opened the pull request:
Updated Sequence Runner have not run tests.

ysoo committed:
Start adding sequence runner tests

ysoo committed:
Updated Sequence Runner

Am not done with tests

ysoo committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into SequenceRunner

jkoritzinsky commented on WPLGSS.UnitTests/SequenceRunnerTests.cs:
```diff
@@ -0,0 +1,45 @@
+            var LabJack = A.Fake<ILabJackGateway>();
+            var configService = A.Fake<IConfigService>();
+            var config = A.Fake<Config>();
+            var service = new DataService(configService, LabJack);
```
Make this an `A.Fake<IDataAquisition>()` instead of `new DataService`.

jkoritzinsky commented on WPLGSS.UnitTests/SequenceRunnerTests.cs:
```diff
@@ -0,0 +1,45 @@
+                await Task.Delay(15);
+            }
+            // How do I verify that the data acquisition service gets called?
+            Assert.True(called);
```
When you make the data service a fake, you can verify that a specific call must have happened.

You can see documentation on how to do that here: http://fakeiteasy.readthedocs.io/en/stable/assertion/

jkoritzinsky commented on WPLGSS/Services/SequenceRunner.cs:
```diff
@@ -12,41 +12,74 @@
+                            Service.SetChannelValue(channel, 1);
+                            break;
+                        case AbortCondition abort:
+                            timer.Start();      // Restart Timer ?
```
You don't need to re-start the timer. You just need to set `sequenceStartTime` to the current time.

jkoritzinsky commented on WPLGSS/Services/SequenceRunner.cs:
```diff
@@ -55,12 +88,31 @@ private void OnTimedEvent(object source, System.Timers.ElapsedEventArgs e)
+        private void Service_ChannelValueUpdated(object sender, ChannelValueUpdatedEventArgs e)
+        {
+            
+            if(AbortCondition.ThresholdMax <= e.Value || AbortCondition.ThresholdMin >= e.Value)
```
We need to support multiple abort conditions. I suggest using a list and checking if any of the conditions are violated.

jkoritzinsky commented on WPLGSS/Services/SequenceRunner.cs:
```diff
@@ -55,12 +88,31 @@ private void OnTimedEvent(object source, System.Timers.ElapsedEventArgs e)
+
+        public void Dispose()
+        {
+            ((IDisposable)timer).Dispose();
```
You shouldn't need this cast here.

ysoo committed:
Sequence Runner fixed

Test not yet implemented

ysoo committed:
Tests

Don't know how to call service.SetChannelValue() with mutable argument :
Channel. Tried subscribing to event channelValueUpdated but think I'm
not doing it right cause didn't pass test.

jkoritzinsky committed:
Cleaned up test and got it to test the sequence runner.
